apps/crud/src/app/app.store.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable, inject } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
+import {
+  ComponentStore,
+  OnStateInit,
+  OnStoreInit,
+  tapResponse,
+} from '@ngrx/component-store';
+import { Observable, exhaustMap } from 'rxjs';
+
+import { Todo } from './interfaces/Todo';
+import { TodoService } from './service/todo.service';
+
+export interface AppState {
+  todos: Todo[];
+  callState: string;
+}
+
+// maybe better to add {providedIn: root} to Injectable
+// is it really necessary to create another store just for a presentational component?
+// pass update / delete methods to TodoItemComponent?
+@Injectable()
+export class AppStore
+  extends ComponentStore<AppState>
+  implements OnStateInit, OnStoreInit
+{
+  // making todo$ and callState$ -> private -> makes testing more difficult ?
```

Comment:
To test private method you can always do:
`(todoService as any)['myPrivateMethod']` 
but this is not very nice indeed